homeassistant/components/statistics/sensor.py

```diff
@@ -264,6 +277,26 @@ def _purge_old(self):
                 self.min_age = self.max_age = dt_util.utcnow()
                 self.change = self.average_change = STATE_UNKNOWN
                 self.change_rate = STATE_UNKNOWN
+        self.async_schedule_update_ha_state()
```

Comment:
This is weird. You're calling `async_schedule_update_ha_state` INSIDE the `async_update` function. An update is already in progress, this is not necessary.

Reply:
OK, understood.